### PR TITLE
[DOCS] Added section that coordinates can be accessed via indexes

### PR DESCRIPTION
**Pull request by Ashafix**
Coulnd't find any documentation that vector coordinates can be accessed like arrays. Don't know what's the best place to add it, I think it should be part of the documentation since it makes writing code much easier.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
Could you give an example? Does this mean v[0] instead of v.x? If so, whats the improvement? :)

**Comment by Ashafix** [comment]
Yes, `v.x` is equivalent to `v[0]`. The 1st one is more readable, the 2nd one has the advantage that code can be reused more easily, e.g. a function which gets an array of vectors as input and modifies each coordinate randomly. If we use `x`, `y`, `x`, we need a different function for `Vector2` and `Vector3`. If we use indexes we can loop until the dimension is reached. Accessing coordinates by index is also common in other languages, e.g. Vex or GLSL. Documenting this feature could help users coming from other languages.

File: doc/classes/Vector2.xml [hunk 1]
```diff
@@ -259,10 +259,10 @@
 	</methods>
 	<members>
 		<member name="x" type="float" setter="" getter="">
-			The vector's x component.
+			The vector's x component. Also accesible by using the index position `[0]`.
```
Comment (akien-mga) [inline doc/classes/Vector2.xml]: You can't use Markdown here, it's a pseudo BBCode. So you should use `[code][0][/code]`. Also typo in "acces**s**ible".

Comment (Ashafix) [reply doc/classes/Vector2.xml]: fixed the Markdown and the typo

**Comment by akien-mga** [comment]
Thanks. Could you squash commits together? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7.

**Pull request merged**